### PR TITLE
Removal of Maxcaps via cvar

### PR DESCRIPTION
**Pull request by TurboTrackerss14**
## About the PR

Praise be community feedback. Let the world be free of funny atmos bombs. 
Single line comment out of the tankexplosion trigger.
## Why / Balance

Usecase is so small that its almost always used to commit rulebreaks rather than reasonable usage. 
- Traitor without DAGD removes entire department(s) 
- Non-antag demolishes security for the tiniest slight 
- "accidental" maxcap creation disregarding the fact its a few specific steps to create. 
- painful optimization on the side of the player to create maxcap "grenades" to eliminate all risk to self. 

While I do wish for there to be a cleaner and more efficient change to maxcaps.
Until such a change is proposed and implimented, this is fit for purpose.
## Technical details

No more gastank explosions is likely to cause issues down the line for pipe bursting etc without being a maxcap. 
## Media

no media.
## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- remove: Gas tank explosions ("max caps") have a reduced range on WizDen servers until they can be reworked into a proper game mechanic.

### CODE REVIEW

**Comment by deepdarkdepths** [comment]
Holy based

**Comment by Just-a-Unity-Dev** [comment]
Can this just be a cvar? I know many servers that would be very fine with having tankbombs enabled

**Comment by mirrorcult** [comment]
why are you doing things like this with comments, add a cvar

**Review by EmoGarbage404 — DISMISSED** [review]
Make it a CVAR and have the logic escape out early so we're not doing atmos calculation for no reason.

File: SpaceStation14.sln [hunk 1]
```diff
@@ -68,39 +68,39 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Robust.Shared.Scripting", "
...
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Robust.Client.NameGenerator", "RobustToolbox\Robust.Client.NameGenerator\Robust.Client.NameGenerator.csproj", "{3CFEB7DB-12C6-46F3-89FC-1450F3016FFA}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Robust.Client.NameGenerator", "RobustToolbox\Robust.Client.NameGenerator\Robust.Client.NameGenerator.csproj", "{3CFEB7DB-12C6-46F3-89FC-1450F3016FFA}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Robust.Client.Injectors", "RobustToolbox\Robust.Client.Injectors\Robust.Client.Injectors.csproj", "{8922428F-17C3-47A7-BFE9-570DEB2464DA}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Robust.Client.Injectors", "RobustToolbox\Robust.Client.Injectors\Robust.Client.Injectors.csproj", "{8922428F-17C3-47A7-BFE9-570DEB2464DA}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "XamlX.IL.Cecil", "RobustToolbox\XamlX\src\XamlX.IL.Cecil\XamlX.IL.Cecil.csproj", "{16F7DE32-0186-44B9-9345-0C20D1BF2422}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "XamlX.IL.Cecil", "RobustToolbox\XamlX\src\XamlX.IL.Cecil\XamlX.IL.Cecil.csproj", "{16F7DE32-0186-44B9-9345-0C20D1BF2422}"
```
Comment (0x6273) [inline SpaceStation14.sln:77]: Don't do whatever this is

File: Content.Server/Atmos/EntitySystems/GasTankSystem.cs [hunk 2]
```diff
@@ -51,8 +55,19 @@ public override void Initialize()
...
+        {
+            base.Shutdown();
+            _cvar.UnsubValueChanged(CCVars.AtmosTankFragment, UpdateFragment);
+        }
+
+        private void UpdateFragment(bool value)
+        {
+            _tankFragmentEnabled = value;
+        }
         private void OnGasShutdown(Entity<GasTankComponent> gasTank, ref ComponentShutdown args)
```
Comment (0x6273) [inline Content.Server/Atmos/EntitySystems/GasTankSystem.cs:71]:
```suggestion
        }
        
        private void OnGasShutdown(Entity<GasTankComponent> gasTank, ref ComponentShutdown args)
```
Always a blank line between methods.

File: Content.Shared/CCVar/CCVars.cs [hunk 3]
```diff
@@ -1240,6 +1240,12 @@ public static readonly CVarDef<bool>
         public static readonly CVarDef<float> AtmosHeatScale =
             CVarDef.Create("atmos.heat_scale", 8f, CVar.SERVERONLY);
 
+        /// <summary>
+        ///     Toggle for Atmos tank canister explosions
+        /// </summary
+        public static readonly CVarDef<bool> AtmosTankFragment =
+            CVarDef.Create("atmos.tank_fragment", true, CVar.SERVERONLY);
```
Comment (0x6273) [inline Content.Shared/CCVar/CCVars.cs:1247]: Should be a cap on radius instead of a bool.

**Review by 0x6273 — CHANGES_REQUESTED** [review]

File: SpaceStation14.sln (hunk 1 above)
Comment (TurboTrackerss14) [reply SpaceStation14.sln:77]: I fucking hate Visual Studio.

File: Resources/ConfigPresets/WizardsDen/wizardsDen.toml [hunk 4]
```diff
@@ -11,6 +11,7 @@ panic_bunker.enable_without_admins = true
 panic_bunker.show_reason = true
 panic_bunker.custom_reason = "You have not played on a Wizard's Den server long enough to connect to this server. Please play on Wizard's Den Lizard, Leviathan, or Farm Grass Hopper until you have more playtime."
 
+
 [infolinks]
```
Comment (0x6273) [inline Resources/ConfigPresets/WizardsDen/wizardsDen.toml:15]: There shouldn't be another blank line here? The other sections just have one blank line between them.

**Comment by Ilya246** [comment]
alternatively we could give them a special explosion damage type which does no structural and can't space floor, plus give it lower damage against people as well

File: Content.Shared/CCVar/CCVars.cs (hunk 3 above)
Comment (0x6273) [reply Content.Shared/CCVar/CCVars.cs:1247]: (you could treat 0 like disabled and still exit out early from atmos code)

**Comment by 0x6273** [comment]
> alternatively we could give them a special explosion damage type which does no structural and can't space floor, plus give it lower damage against people as well

This should be configurable in that case I think. But just limiting radius is probably good enough.

**Comment by cheltyi** [comment]
yeah, nerf everything but buff security, we shouldn't have fun in a round

**Comment by 0x6273** [comment]
> remove: Gas Tank explosions have been removed. Including but not limited to maxcaps.

This changelog could be a bit misleading for downstreams since they are actually enabled by default, and only disabled in the wizden server config.

**Comment by 0x6273** [comment]
As a game admin, I've had instances where I wish I could disable maxcaps temporarily, like I see atmos making a maxcap and I don't know what they intend to do with it. Like they could be a raider or maybe they just want to explode it in space for fun?

I think having a cvar for this can be very useful, but since disabling maxcaps by default is so controversial I think that should be done in a separate pr instead.

**Comment by cheltyi** [comment]
no fun allowed

**Comment by Killerqu00** [comment]
> yeah, nerf everything but buff security, we shouldn't have fun in a round
> 
> just a simple "No fun allowed"

reading comprehension 0 + security and command gets overwhelmed 9/10 rounds already :/

**Comment by Ilya246** [comment]
i've said this on discord before, but repeating this here:

this PR will prevent atmosians from making maxcaps; however, the syndies (or even the self-antags) doing it aren't going to go anywhere, so they will just do other things<br>i suspect that we will instead get plasmafloods or stronger, which can and likely will be be even worse than maxcaps via being essentially always evac-worthy and affecting *everyone* and not a particular area if done properly

therefore, i wish for us to take into consideration that this PR will likely **not** cause the following changes, cause them to a lesser extent than expected, or even worsen the related situations:
- self-antags causing mass death
- lower admin pressure
- lower sec pressure
- make non-DAGD atmos traitors cause mass death

**Comment by LankLTE** [comment]
"removing maxcaps is a bad idea because then antags will just do something WORSE" is a bad argument because then we can just.. also nerf those worse things?

**Comment by cheltyi** [comment]
> "removing maxcaps is a bad idea because then antags will just do something WORSE" is a bad argument because then we can just.. also nerf those worse things?

Yeah, as I said before, no fun allowed, buff security, nerf everything

**Comment by Ilya246** [comment]
> "removing maxcaps is a bad idea because then antags will just do something WORSE" is a bad argument because then we can just.. also nerf those worse things?

then do nerf them?

**Comment by ScarKy0** [comment]
If we just nerf everything at once then the argument "no fun allowed" makes sense at that point

**Comment by cheltyi** [comment]
Big part of community hate, when something gets nerfed cuz crying sec, which can't just stunmeta them

**Comment by Ilya246** [comment]
besides, even if you do nerf all of atmos, people will still be able to singuloose

and, once again, the atmos traitors you deprived of their past options will not go anywhere, they will just move on to doing other things, like singuloosing

maxcaps are one of the more tame options

**Comment by Ilya246** [comment]
i've thought about it a bit more, and come to a conclusion that this PR might be treating a symptom and not the cause

what is the cause, you ask? so, have you ever seen anyone get arrested for *making* maxcaps, as opposed to using them? no? here's your answer<br>personally, i've had CEs and even captains walk by maxcap making setups while knowing what they are, maybe be at most a little concerned, then leave without doing anything

i think the issue is that for some reason command, sec, and other atmosians don't do *anything* even when it's known one atmos is making a maxcap, even though the setups for it are rather conspicuous and can even be distinguished by people that don't know how to make maxcaps

**Comment by shampunj** [comment]
No

**Comment by cheltyi** [comment]
No

**Comment by Plykiya** [comment]
> i've thought about it a bit more, and come to a conclusion that this PR might be treating a symptom and not the cause
> 
> what is the cause, you ask? so, have you ever seen anyone get arrested for _making_ maxcaps, as opposed to using them? no? here's your answerpersonally, i've had CEs and even captains walk by maxcap making setups while knowing what they are, maybe be at most a little concerned, then leave without doing anything
> 
> i think the issue is that for some reason command, sec, and other atmosians don't do _anything_ even when it's known one atmos is making a maxcap, even though the setups for it are rather conspicuous and can even be distinguished by people that don't know how to make maxcaps

I have like 1400 hours in this game and I still don't know how to identify a maxcap setup

If you know what a maxcap setup you are not likely to snitch on the person doing it either, because atmos has a cult of knowledge around creating WMDs just because they can as a flex and other atmos techs respect the art

**Comment by Emisse** [comment]
I think pretending that max caps as they are, is ok, is hilariously insane. They either need to be reigned in to mini bomb sizes or removed. This is coming from the person who unnerfed them like two years ago.

**Comment by BasedUser** [comment]
> This is coming from the person who unnerfed them like two years ago.

[you mean eclips_e](https://github.com/space-wizards/space-station-14/commit/21fd9ddacf2fb85c0536461dc1a1e4b87fc09843)

**Comment by Emisse** [comment]
read the pr you just linked

**Comment by Golinth** [comment]
> > This is coming from the person who unnerfed them like two years ago.
> 
> [you mean eclips_e](https://github.com/space-wizards/space-station-14/commit/21fd9ddacf2fb85c0536461dc1a1e4b87fc09843)

image

**Comment by Ilya246** [comment]
> I think pretending that max caps as they are, is ok, is hilariously insane. They either need to be reigned in to mini bomb sizes or removed. This is coming from the person who unnerfed them like two years ago.

yes so nerf them instead of removing them, that's what my entire point earlier was about, that would avoid many of the issues

my suggestion earlier was to:
- make them not deal structural
- make them unable to space station floor
- make them deal less damage overall

**Comment by BasedUser** [comment]
> image

o h

**Comment by kotav1322** [comment]
> > I think pretending that max caps as they are, is ok, is hilariously insane. They either need to be reigned in to mini bomb sizes or removed. This is coming from the person who unnerfed them like two years ago.
> 
> yes so nerf them instead of removing them, that's what my entire point earlier was about, that would avoid many of the issues
> 
> my suggestion earlier was to:
> 
> * make them not deal structural
> * make them unable to space station floor
> * make them deal less damage overall

They are perfect in the way they are now

**Comment by kotav1322** [comment]
> at this point in favor of removing it solely because people are going so far as to use calculators to min-max their yields instead of just playing the fuckin game and figuring it out either through in-game discussions or trial and erroring shit
> 
> put the ti84s away ya fuckin dorks
> 
> EDIT: still downvoting this PR to contribute to it being the most disliked PR on the repo's history though

Kill yourself no fun allowing nerd. At this point, remove the ability to enable fucking harm mode for people that are not sec or antag

**Comment by EthanQix** [comment]
In all my rounds I've seen a maxcap explode once, and that was used against a nukie team. Maxcaps are a non issue. This PR is definitely in the No Fun Allowed category.

**Comment by retequizzle** [comment]
> > at this point in favor of removing it solely because people are going so far as to use calculators to min-max their yields instead of just playing the fuckin game and figuring it out either through in-game discussions or trial and erroring shit
> > put the ti84s away ya fuckin dorks
> > EDIT: still downvoting this PR to contribute to it being the most disliked PR on the repo's history though
> 
> Kill yourself no fun allowing nerd. At this point, remove the ability to enable fucking harm mode for people that are not sec or antag

that's rich coming from the human equivalent of a participation trophy, opinion's about as useful as pedals on a wheelchair. get fucked and stay fucked lol

**Comment by slarticodefast** [comment]
Stay civil please. Feel free to discuss this using proper arguments, we read them and will consider them in the final decision. But harassing each other does not help in any way.

**Comment by JIPDawg** [comment]
Just putting in my anecdotal experience here. In my ~500 hours of game play I can say with like 95% certainty that I have seen a maxcap _maybe_ 1 time ever.

**Comment by HenryOfSkalitz** [comment]
Maximum Capacity - also known as MaxCap - is the most powerufl bomb setup, intended to be used by antagonists and/or to fight antagonist in a good version of the game (Space Station 13), and since it's very first iteration did just that.

Inability for Administration to control and/or prevent and/or manage consenquences of "grief" should not be a reason to prevent a game mechanic to be removed.

If you plan, or wish to, remove something, you must reason it with "It breaks the game by itself", or "It abuses even within rule realm"

The only time reasoning of "People that break rules only do that" is applicable to actual exploits, bugs, and unintended mechanics.

Ability to make explosions using the oldest system of entire game, ported to its "successor" as SS14 proudly calls itself being removed thanks to these reasons brought in PR are silly. here's step by step reason.

"Well non-antags explode security for the tiniest slight!"
Well, non-antags also beat security, vandalize Brig, rob Armory and walk around with guns for the tiniest slight. Is the issue in max caps?

"Well traitors without DAGD removes entire department(s)"
Bombs can be used to cause diversion of attention, sabotage specific places (T-Comms, AI satellite, SECURITY BRIG filled to the brim with guns used to kill people, MEDICAL where wounded target and/or security and/or command are brought, etc etc, and creative use of very devastating and long-made explosive is beyond reason. Traitors already have ability to buy other type of explosives, here and now, but can take risky attempt to make manual bomb.

"Well people "accidentally" create maxcap disregarding the fact its a few specific steps to create!"
"I accidentally crashed the server with very specific exploit with very specific steps. Please remove bug and not me!" is something you'd also believe in.

"painful optimization on the side of the player to create maxcap "grenades" to eliminate all risk to self" is probably somewhat reasonable argument, but towards fixing lack of risk to self rather than entire removal of entire feature.

Skill issue.

**Comment by lzk228** [comment]
tbh i also have seen like one or maybe even zero maxcaps in 3000 hours of playtime
and i don't really understand what is the issue here

**Comment by Golinth** [comment]
I used to see them a bit before they got super nerfed, but if they are never seen, why not remove them. They cause more problems than not, and like the PR states, is against the rules in nearly all scenarios

**Comment by Errant-4** [comment]
**IF** the actual frequency of maxcap explosions factors into this decision, it would probably not be difficult to add logging to atmos/explosion calculations so that every time something explodes over value X it leaves a log entry. Then after a week or two we could actually see in the database how common these things are

**Comment by Savsj** [comment]
So, my one experience with max caps was a traitor atmos tech made max caps and RRed more than 12 people by deleting med, sec, hops office, and cargo AND ruined the event the admin was running in under 2 minutes of throwing bombs. These are NOT hard to make, they require no skill just following a step by step guide and are only a single step above loosing in terms of creativity. The traitor in question didnt even have DAGD but theyd only targeted individuals they needed to technically so thry didnt break any rules, they did however ruin the round. So maxcaps either need a nerf or removed.

**Comment by EthanQix** [comment]
> So, my one experience with max caps was a traitor atmos tech made max caps and RRed more than 12 people by deleting med, sec, hops office, and cargo AND ruined the event the admin was running in under 2 minutes of throwing bombs. These are NOT hard to make, they require no skill just following a step by step guide and are only a single step above loosing in terms of creativity. The traitor in question didnt even have DAGD but theyd only targeted individuals they needed to technically so thry didnt break any rules, they did however ruin the round. So maxcaps either need a nerf or removed.

"This happened once" is not an argument. Rounds get ruined regularly by unlikely, random bullshite.

If maxcaps happened to ruin the whole round, without technically breaking rules, *with any kind of regularity*, then it would be an actual issue.

**Comment by whateverusername0** [comment]
mald pr-

**Comment by cfif126** [comment]
mid pr

**Comment by Erisfiregamer1** [comment]
> So, my one experience with max caps was a traitor atmos tech made max caps and RRed more than 12 people by deleting med, sec, hops office, and cargo AND ruined the event the admin was running in under 2 minutes of throwing bombs. These are NOT hard to make, they require no skill just following a step by step guide and are only a single step above loosing in terms of creativity. The traitor in question didnt even have DAGD but theyd only targeted individuals they needed to technically so thry didnt break any rules, they did however ruin the round. So maxcaps either need a nerf or removed.

this is like saying meteors should be removed because they sometimes destroy evac.

**Comment by Golinth** [comment]
> this is like saying meteors should be removed because they sometimes destroy evac.

Except someone is working on meteor defenses to stop meteors from hitting vital areas before the station is ready, because they recognize it as a problem. No one is saying maxcaps can’t be reintroduced in a better form in the future, but as they are now they are unbalanced rule breaking devices 99 times outta 100.

**Comment by Ilya246** [comment]
> So, my one experience with max caps was a traitor atmos tech made max caps and RRed more than 12 people by deleting med, sec, hops office, and cargo AND ruined the event the admin was running in under 2 minutes of throwing bombs. These are NOT hard to make, they require no skill just following a step by step guide and are only a single step above loosing in terms of creativity. The traitor in question didnt even have DAGD but theyd only targeted individuals they needed to technically so thry didnt break any rules, they did however ruin the round. So maxcaps either need a nerf or removed.

how long ago was that? maxcaps have been directly or indirectly nerfed numerous times, so anything before 2024 was with maxcaps that were considerably stronger

**Comment by JIPDawg** [comment]
> > this is like saying meteors should be removed because they sometimes destroy evac.
> 
> Except someone is working on meteor defenses to stop meteors from hitting vital areas before the station is ready, because they recognize it as a problem. No one is saying maxcaps can’t be reintroduced in a better form in the future, but as they are now they are unbalanced rule breaking devices 99 times outta 100.

First that someone has to finishing reworking salv, then maybe work on revenant\xenobiology\xenoarcheology...etc. what I'm trying to say is that I think it's low in the priority and won't be seen anytime soon. (Meteor defense, that is) I'm not laying blame at all id rather everything else they are working on first personally lol .

File: Content.Server/Atmos/EntitySystems/GasTankSystem.cs [hunk 5]
```diff
@@ -51,6 +55,18 @@ public override void Initialize()
             SubscribeLocalEvent<GasTankComponent, GasAnalyzerScanEvent>(OnAnalyzed);
             SubscribeLocalEvent<GasTankComponent, PriceCalculationEvent>(OnGasTankPrice);
             SubscribeLocalEvent<GasTankComponent, GetVerbsEvent<AlternativeVerb>>(OnGetAlternativeVerb);
+            _cvar.OnValueChanged(CCVars.AtmosTankFragment, UpdateFragment, true);
+        }
+
```
Comment (metalgearsloth) [inline Content.Server/Atmos/EntitySystems/GasTankSystem.cs:60]: There's a helper method for this, subs.something.

File: Content.Server/Atmos/EntitySystems/GasTankSystem.cs [hunk 6]
```diff
@@ -51,6 +55,18 @@ public override void Initialize()
...
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            _cvar.UnsubValueChanged(CCVars.AtmosTankFragment, UpdateFragment);
+        }
+
```
Comment (metalgearsloth) [inline Content.Server/Atmos/EntitySystems/GasTankSystem.cs:66]: Unnecessary.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Not really sure why it's a bool and not a float.

**Comment by EthanQix** [comment]
> > this is like saying meteors should be removed because they sometimes destroy evac.
> 
> Except someone is working on meteor defenses to stop meteors from hitting vital areas before the station is ready, because they recognize it as a problem. No one is saying maxcaps can’t be reintroduced in a better form in the future, but as they are now they are unbalanced rule breaking devices 99 times outta 100.

The meteor active defense need to happen because current meteors are unfun - you're just warned that you're gonna take meteors to the face and there's nothing you can do to prevent it. The only agenda players have is repairing the (potentially extensive) damage. And meteors happen several times per round.

On the opposite side of the balance, you have maxcaps. While being able to do massive damage, they only happen once in a blue moon and are an entirely player-controlled mechanic. Having a maxcap "ruin" a round is a player-controlled event in itself. This PR is entirely unneeded.

**Comment by TurboTrackerss14** [comment]
Please hold while I change every thing to sub.cvar wahoo.

File: Content.Server/Atmos/EntitySystems/GasTankSystem.cs (hunk 5 above)
Comment (TurboTrackerss14) [reply Content.Server/Atmos/EntitySystems/GasTankSystem.cs]: Thy will is done.

**Comment by TurboTrackerss14** [comment]
Subs.Cvar complete. We game.

File: Content.Shared/CCVar/CCVars.cs [hunk 7]
```diff
@@ -1240,6 +1240,12 @@ public static readonly CVarDef<bool>
         public static readonly CVarDef<float> AtmosHeatScale =
             CVarDef.Create("atmos.heat_scale", 8f, CVar.SERVERONLY);
 
+        /// <summary>
+        /// Max range for explosions caused by a gas reaction.
+        /// </summary>
+        public static readonly CVarDef<float> AtmosTankFragment =
+            CVarDef.Create("atmos.max_explosion_Range", 26f, CVar.SERVERONLY);
```
Comment (0x6273) [inline Content.Shared/CCVar/CCVars.cs:1247]:
```suggestion
            CVarDef.Create("atmos.max_explosion_range", 26f, CVar.SERVERONLY);
```
And update config

**Comment by TurboTrackerss14** [comment]
>Since we are on the topic of nerfing maxcaps, how about something like reducing range, increasing damage, and reducing structural damage or something like that?
Point 4 why/balance

>Also "accidental" maxcaps are the same as "accidentally" connecting plasma to distro, and for maxcaps is even harder, i doubt theres a single person that has accidentally maxcapped, and i dont think it should be taken into account as a reason to nerf them
Yes thats why it was in big quotes, its "accidentally" not accidentally. The classic "I didn't know tanks blew up I just wanted to make a hot canister what"

>How about something like the smaller the range the more damage they do? it wouln't make much sense tho, but from a gameplay perspective, maybe it could help with those issues?
Point. 4. Why/Balance.

**Comment by Cojoke-dot** [comment]
I need to get a DAGD to show people why this is an issue… maxcaps are exclusively used for mass murder, as per the point of a massive bomb. Sure, nerf the damage and range then… what? Nobody uses it because it's still a secret people keep that's not all that strong as strong as an bomb that chem can make that's much easier to identify? Nobody likes getting hit by a big bomb that kills them and nobody likes wimpy bombs that do not affect anyone. There is no counter for an atmos walking into med with a tank and a canister. 0 sound, a timer long enough for the person deploying it to get away, a tossable projectile.

I have 2 ideas that could replace them:
1. A randomization to the formula so Atmos has to test and it can't be made 5:00 into the shift
2. Maxcaps go “poof”, rapidly filling a room with the gas in the canister. Rapid sleep bombs, plasma canister scare, refill a room with air. More utility than a direct bomb(trigger would have to be different too)

**Comment by 0tito** [comment]
The painful optimizations are not painful, and its the fun part, having the ability to play around with values and get different result is what makes them fun, and like a breath of fresh air after making frezon for the 50th time. (Obviously if someone plays around with the values in game and not on localhost its on them), also, the risk to self part, why should there be a risk to self? if i get dagd, i can plasma flood, hot vapor flood, singuloose, tesloose. If im a chem syndie i can make explosives, lethal foams, deadly poisons. If im a scientist i can make a borg army. All this has almost no risk, maybe the borg army.

All the explosives on the syndicate uplink have almost 0 risk to self, except for getting detected. Wouldn't a change like this have the same effect? If you see an atmos walk, throw a tank and it explodes, since now it has less range, theres a high chance people will witness it and survive, different from actual maxcaps in wich no witness survives.

Also this would make it so that its decently dodgeable, like grenades, and it would force atmos players to use those "painful" optimizations to tune the bombs to have fewer ticks till explosion, since it would be harder to aim at/hit people with less range.

Also Cojoke-dot, if they were weaker than chem bombs then theres no reason to remove them, being weaker shoulnt be a reason to just remove it, and what im saying is to nerf the range but buff the damage, so it woulnt be a wimpy bomb. 

Also yeah, people dont like being blown up, but explosives are a thing in the game, unless we remove all explosives & meteors, dying to an explosion being boring is not a reason for  removing just maxcaps (if they had the range nerfed, how they are right now, yeah, its too big you cant even see it and you die)

So what i propose is just turning a weapon of mass destruction into a tool you can use to target specific people and kill them.

Also, for your point 2, tanks have only 5l and for rapidly filling the room with the content of the canister i can just open it, it would basically be the same as removing maxcaps, cause if i want to fill a room with the contents of a canister i can alredy do that

**Comment by 0tito** [comment]
>Turbo is referring to people making programs to maximize time - power. Get enough time to get away while dealing enough damage to remove a department. Removal of the risk that is meant to be inherent with using maxcaps through optimization. I figured out maxcaps with a tip here and there and did experiments. Fun times in atmos till I almost loosed the singularity and blew up the teg in one shift 🙃

I mean thats voluntary work, he made that program cause he wanted, i doubt he felt it was painful, and some atmos stuff requires calculations, i dont think this is an issue. 
Also testing maxcaps in the middle of atmos in a normal round is on you, if you are going to test explosives, test them either in a platform in space, or in a localhost.

>Its a tank, if someone throws a tank at you, you are not going to go into evasive maneuvers and flee for your life

>They already one shot people, how much more would you wan to buff the damage? Craftable mini bombs that deal 200 blunt and have no beep?

We could add sprite once the tank reaches more than 1500kpa or something, and maybe a sound when it reaches 3000kpa+

Also, yeah maxcaps one shot people, but smaller atmos bombs dont, the idea would be that smaller atmos bombs also kill, to discourage the use of bigger radius.

>Yes… none of the bombs are silent, disguised, and have a meaningful radius. The most you have is a pen bomb which is just a c4. 1/5 of your TC for one bomb compared to the infinite you can make at atmos

Well, chem can also mass produce bombs, 0 tc cost, that instakill, but with a smaller range, and chemicals that round remove people.

>Maybe it can force a bit of the gas into people's lungs who have internals on and will not have a cap on pressure. It would be only useful in larger rooms.

Sounds cool honeslty, but i feel like it would only be useful if we were to add other gasses that have different effect when inhaled, right now, only gasses that would be useful to force inject into people are n2o and frezon, and if im going to make all the people in a room be in contact with a gas i made, first, it would only be useful for mass murder anyways, even more now with the cold slowdown nerf (cause before the cold was useful if you injected yourself with cryox).

Second, why frezon or n2o flood a room when i can flood it with hot vapor and insta kill everyone? Why force people to breathe n2o when they can wake up and toggle internals, instead of flooding with n2o its simply better to flood with frezon, wich will also make n2o and it has the benefit of also dealing damage to people that dont have internals. So you'd only flood a room with either frezon, or superheated gasses, both will end in mass murder, just that the hot gas will also round remove

Now, i cant deny that these changes would take more work than making maxcaps a cvar, to turn them to 0 and kill them, but i think it would be a healthier change. But i guess the answer i will get is wyci :godo:

**Comment by Partmedia** [comment]
@TurboTrackerss14 I pushed some minor nitpicky changes directly to your branch. I'm not super sure how Subs.CVar works, but does this correctly call `UpdateMaxRange()` on init when the CVar is set to something else (e.g. 0) in the config file? If you can't look into this I'll take a look at this a bit later.

**Comment by TurboTrackerss14** [comment]
> @TurboTrackerss14 I pushed some minor nitpicky changes directly to your branch. I'm not super sure how Subs.CVar works, but does this correctly call `UpdateMaxRange()` on init when the CVar is set to something else (e.g. 0) in the config file? If you can't look into this I'll take a look at this a bit later.

Changes look fine to me 
(Fuck I forgot to delete the shutdown method entirely aswell thanks for that)

**Comment by TurboTrackerss14** [comment]
I blindly trusted the comments from contrib-voice during my shitcode fixing stream. 
Subs.CVar is a magical entity to me that I made work. 

If you are willing to prettify my code I would be more than allowing of it.

File: Content.Server/Atmos/EntitySystems/GasTankSystem.cs [hunk 8]
```diff
@@ -51,6 +55,12 @@ public override void Initialize()
             SubscribeLocalEvent<GasTankComponent, GasAnalyzerScanEvent>(OnAnalyzed);
             SubscribeLocalEvent<GasTankComponent, PriceCalculationEvent>(OnGasTankPrice);
             SubscribeLocalEvent<GasTankComponent, GetVerbsEvent<AlternativeVerb>>(OnGetAlternativeVerb);
+            Subs.CVar(_cfg, CCVars.AtmosTankFragment, UpdateMaxRange);
```
Comment (Partmedia) [inline Content.Server/Atmos/EntitySystems/GasTankSystem.cs:58]:
```suggestion
            Subs.CVar(_cfg, CCVars.AtmosTankFragment, UpdateMaxRange, true);
```
This needs the `invokeImmediately` argument set to true in order to correctly update at the beginning.

**Comment by TurboTrackerss14** [comment]
Looks to be all set 🫡

**Review by Partmedia — APPROVED** [review]

File: Resources/ConfigPresets/WizardsDen/wizardsDen.toml [hunk 9]
```diff
@@ -38,3 +38,6 @@ see_own_notes = true
 deadmin_on_join = true
 new_player_threshold = 600
 alert.min_players_sharing_connection = 2
+
+[atmos]
+max_explosion_range = 0
```
Comment (slarticodefast) [inline Resources/ConfigPresets/WizardsDen/wizardsDen.toml:43]:
```suggestion
max_explosion_range = 5
```

**Comment by slarticodefast** [comment]
Hey everyone, we had a very long discussion and multiple votes with the maintainers and admins. We came to the conclusion that max caps are some kind of tradition from SS13 and an interesting result of our atmos simulation, that deserves to be kept in some form in the future. However, due to problems with their current mechanics and the huge burden on our game admins, we have decided to limit their explosion radius using a cvar until we can rework some of the mechanics behind maxcap production and turn them into a proper feature. Servers that wish to keep them as they currently are can simply change that cvar.

Current problems:
- From a WizDen server rule perspective the DAGD objective is currently the only acceptable reason to maxcap.
- Even more problematic are the timed, easily mass-producable minicap grenades with smaller radius, that allow the user to run of with zero risk to themselves, allowing players to do massive station destruction.
- If we were to reduce the cvar to anything high enough so that it would not affect these, it would only further encourage their mass production.
- Most players making maxcaps simply follow a recipe without doing their own research and experimentation. From a gameplay perspective atmos should be more than a list of steps you have to follow.
- The steps to make a maxcaps are unknown enough so that many players don't notice it, even when done in plain sight and there is no warning or way to stop the bomb.

What a good solution for a rework should include:
- They should be turned into a proper game mechanic with a need for experimentation and a chance of failure.
- They should be properly telegraphed, so there is a way to stop the bomber.
- Their production should not be easily repeatable using a recipe and they should not be mass producible.

Some ideas we will investigate:
- Make the canister the exploding part instead of the tank. This means a lot more gas will be needed for a single bomb and they cannot be thrown.
- Randomly vary the heat capacities of gases by a few percent each round. This won't be noticable for normal gameplay, but will ensure that maxcaps are a simple recipe that you cannot look up. Instead the player would have to do some experiments and apply the scientific method to find out the current values, then adjust the recipe.
- Make them require more items than tanks and canisters. This could be a piping setup at the location of the explosion or a traitor only item.

**Comment by deltanedas** [comment]
> Randomly vary the heat capacities of gases by a few percent each round. This won't be noticable for normal gameplay, but will ensure that maxcaps are not a simple recipe that you can look up somewhere. Instead the player would have to do some experiments and apply the scientific method to find out the current values, then adjust the recipe.

would just mean measure some gases heat capacity at the start then recalculate the ratios, gatekeeping it a bit but thats all

**Comment by cfif126** [comment]
Womp womp

**Comment by deepdarkdepths** [comment]
Yeeess siiiiirrrr

**Comment by VasilisThePikachu** [comment]
I updated the title for **server owners** (well the commit already made it in now soo)

If you don't like this change for your **FORK** you don't have to revert this pr. 

The default value is the same as before, only wizards den servers are effected

**Pull request merged**